force_bdss/data_sources/base_data_source_model.py

```diff
@@ -46,43 +46,218 @@ class BaseDataSourceModel(BaseModel):
     #: this and adapt the visual entries.
     changes_slots = Event()
 
-    def verify(self):
-        """ Verify the data source model.
+    def __init__(self, factory=None, input_slot_info=None,
```

Comment:
Here we pass in all arguments except `input_slot_info` and `output_slot_info`, so that their default values will be assigned first. The `factory` trait is included explicitly only to allow for backwards compatibility, since it is often passed in as an unnamed first argument,